src/mongo/shell.ts

```diff
@@ -41,28 +52,36 @@ export class Shell {
 			this.disposables.push(toDisposable(() => ee.removeListener(name, fn)));
 		};
 
-		once(this.mongoShell, 'error', result => this.onResult.fire(result));
-		once(this.mongoShell, 'exit', result => this.onResult.fire(result));
-
-		let buffers: string[] = [];
-		on(this.mongoShell.stdout, 'data', b => {
-			let data: string = b.toString();
-			const delimitter = `${this.executionId}${os.EOL}`;
-			if (data.endsWith(delimitter)) {
-				const result = buffers.join('') + data.substring(0, data.length - delimitter.length);
-				buffers = [];
-				this.onResult.fire({
-					exitCode: void 0,
-					result,
-					stderr: void 0
-				});
+		// tslint:disable-next-line:no-any
+		once(this.mongoShell, 'error', (error: any) => {
+			this.fireOnResult(error);
+		});
+		once(this.mongoShell, 'exit', (exitCode: number) => {
+			console.error("outputdata:" + outputData);
+			console.error("stderr:" + stderrData);
+			this.fireOnResult({ stringResult: undefined, exitCode });
+		});
+
+		let outputData: string = "";
+		let stderrData: string = "";
+		on(this.mongoShell.stdout, 'data', (buffer: Buffer) => {
+			let data: string = buffer.toString();
+			const delimiter = `${this.executionId}${os.EOL}`;
+			if (data.endsWith(delimiter)) {
+				const result: string = outputData + data.substring(0, data.length - delimiter.length);
+				outputData = "";
+				this.fireOnResult({ stringResult: result });
 			} else {
-				buffers.push(b);
+				outputData += data;
 			}
 		});
 
-		on(this.mongoShell.stderr, 'data', result => this.onResult.fire(result));
-		once(this.mongoShell.stderr, 'close', result => this.onResult.fire(result));
+		on(this.mongoShell.stderr, 'data', (buffer: Buffer) => {
+			stderrData += buffer.toString();
+		});
+		once(this.mongoShell.stderr, 'end', () => {
```

Comment:
Looks like the event type was modified from `close` to `end`. What's the effect of this?